src/installer.ts

```diff
@@ -320,17 +324,29 @@ async function resolveVersionFromManifest(
 }
 
 // TODO - should we just export this from @actions/tool-cache? Lifted directly from there
-function evaluateVersions(versions: string[], versionSpec: string): string {
+function evaluateVersions(versions: INodeVersion[], versionSpec: string): string {
   let version = '';
   core.debug(`evaluating ${versions.length} versions`);
   versions = versions.sort((a, b) => {
-    if (semver.gt(a, b)) {
+    const versionA = semver.coerce(a.version);
+    const versionB = semver.coerce(b.version);
+    // If versions are equal, compare date instead
+    if (versionA === versionB || versionA === null || versionB === null) {
+      if (new Date(a.date) > new Date(b.date)) {
+          return 1;
+      }
+      return -1;
+  }
+  if (semver.gt(versionA, versionB)) {
       return 1;
     }
     return -1;
   });
   for (let i = versions.length - 1; i >= 0; i--) {
-    const potential: string = versions[i];
+    const potential: string = versions[i].version;
+    const semverPotential = semver.coerce(potential);
+        if (semverPotential === null)
+            continue;
     const satisfied: boolean = semver.satisfies(potential, versionSpec);
```

Comment:
Hi!

I think `{ includePrerelease: true }` should be added here, otherwise it won't match version numbers like `v12.22.12-rc.0`.